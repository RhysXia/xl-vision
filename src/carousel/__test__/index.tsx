import { mount } from 'enzyme'
import * as React from 'react'
import Carousel from '../index'

describe('Carousel', () => {
  it('基本用法', () => {
    const wrapper = mount(
      <Carousel>
        <div className={'item1'}>0</div>
        <div className={'item2'}>1</div>
        <div className={'item3'}>2</div>
      </Carousel>
    )
    expect(wrapper.render()).toMatchSnapshot()
  })
})
