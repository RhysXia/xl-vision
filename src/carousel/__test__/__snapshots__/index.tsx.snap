// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel 基本用法 1`] = `
<div
  class="xl-carousel xl-carousel--horizontal"
  style="height: 100%; width: 100%;"
>
  <div
    class="xl-carousel__wrap"
  >
    <div
      class="xl-carousel__items"
      style="width: 0px; transform: translate3d(-0px, 0, 0);"
    >
      <div
        class="xl-carousel__item"
        style="width: 0px;"
      >
        <div
          class="item3"
        >
          2
        </div>
      </div>
      <div
        class="xl-carousel__item"
        style="width: 0px;"
      >
        <div
          class="item1"
        >
          0
        </div>
      </div>
      <div
        class="xl-carousel__item"
        style="width: 0px;"
      >
        <div
          class="item2"
        >
          1
        </div>
      </div>
      <div
        class="xl-carousel__item"
        style="width: 0px;"
      >
        <div
          class="item3"
        >
          2
        </div>
      </div>
      <div
        class="xl-carousel__item"
        style="width: 0px;"
      >
        <div
          class="item1"
        >
          0
        </div>
      </div>
    </div>
  </div>
  <button
    class="xl-carousel__arrow xl-carousel__arrow--first"
    style="display: none;"
  >
    <i
      class="xl-icon"
    >
      <svg
        fill="currentColor"
        style="height: 1em; width: 1em;"
        viewBox="0 0 256 512"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </i>
  </button>
  <button
    class="xl-carousel__arrow xl-carousel__arrow--last"
    style="display: none;"
  >
    <i
      class="xl-icon"
    >
      <svg
        fill="currentColor"
        style="height: 1em; width: 1em;"
        viewBox="0 0 256 512"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </i>
  </button>
  <ul
    class="xl-carousel__dots"
  >
    <li
      class="xl-carousel__dot"
    >
      <button
        class="xl-carousel__dot-inner xl-carousel__dot-inner--active"
      >
        0
      </button>
    </li>
    <li
      class="xl-carousel__dot"
    >
      <button
        class="xl-carousel__dot-inner"
      >
        1
      </button>
    </li>
    <li
      class="xl-carousel__dot"
    >
      <button
        class="xl-carousel__dot-inner"
      >
        2
      </button>
    </li>
  </ul>
</div>
`;
